src/argilla/client/feedback/integrations/huggingface/dataset.py

```diff
@@ -112,12 +112,17 @@ def _huggingface_format(dataset: "FeedbackDataset") -> "Dataset":
 
         hf_dataset["metadata"] = []
 
+        empty = {
+            "user_id": [],
+            "value": [],
+            "status": [],
+        }
         for record in dataset.records:
             for field in dataset.fields:
                 hf_dataset[field.name].append(record.fields[field.name])
             for question in dataset.questions:
                 if not record.responses:
-                    hf_dataset[question.name].append(None)
+                    hf_dataset[question.name].append(empty)
```

Comment:
Additionally, this approach is "columnar", e.g.:
```python
{
      "user_id": [..., ..., ...],
      "value": [4, 3, 1],
      "status": ["submitted", "submitted", "discarded"],
}
```
Have we considered a row-based approach?
```python
[{"user_id": ..., "value": 4, "status": "submitted"},
{"user_id": ..., "value": 3, "status": "submitted"},
{"user_id": ..., "value": 1, "status": "discarded"}]
```
With the latter setup, it is much easier to select all values with the right status:
```python
[annotation["value"] for annotation in X if annotation["status"] == "submitted"]
```
where X is the list.

Right now, you have to do something like:
```python
[value for value, status in zip(X["value"], X["status"]) if status == "submitted"]
```

Thoughts?

Reply:
Yes I agree with a row-approach, I also think is better, we would lose consistency in case we want to add a non-matching dict, but I think that even though the `datasets.Features` may be more generic if we rely on Python dicts rather than on `datasets.Sequence`, it's probably better and easier to handle this way